File: app/src/main/java/com/vtrack/data/model/MaintenanceRecord.kt
package com.vtrack.data.model

import androidx.room.Entity
import androidx.room.ForeignKey
import androidx.room.Index
import androidx.room.PrimaryKey

@Entity(
    tableName = "maintenance_records",
    foreignKeys = [
        ForeignKey(
            entity = MaintenanceType::class,
            parentColumns = ["id"],
            childColumns = ["maintenanceTypeId"],
            onDelete = ForeignKey.CASCADE
        ),
        ForeignKey(
            entity = Vehicle::class,
            parentColumns = ["id"],
            childColumns = ["vehicleId"],
            onDelete = ForeignKey.CASCADE
        )
    ],
    indices = [
        Index("maintenanceTypeId"),
        Index("vehicleId"),
        Index("vehicleId", "maintenanceTypeId", "odometer")
    ]
)
data class MaintenanceRecord(
    @PrimaryKey(autoGenerate = true) val id: Long = 0,
    val maintenanceTypeId: Long,
    val vehicleId: Long,
    val date: Long,
    val odometer: Int,
    val cost: Double? = null,
    val notes: String? = null,
    val createdAt: Long = System.currentTimeMillis()
)

File: vtrack-ios/VTrack/Models/MaintenanceRecord.swift
import Foundation
import SwiftData

@Model
final class MaintenanceRecord {
    var maintenanceType: MaintenanceType?
    var vehicle: Vehicle?
    var date: Date
    var odometer: Int
    var cost: Double?
    var notes: String?
    var createdAt: Date

    init(maintenanceType: MaintenanceType, vehicle: Vehicle, date: Date, odometer: Int, cost: Double? = nil, notes: String? = nil) {
        self.maintenanceType = maintenanceType
        self.vehicle = vehicle
        self.date = date
        self.odometer = odometer
        self.cost = cost
        self.notes = notes
        self.createdAt = Date()
    }
}
